File: AtCoder/ABC/abc324/d/main.py
import math
_=input()
s=input()
cnt=[0]*10
for x in s: cnt[int(x)]+=1
lim=int(''.join(sorted(s,reverse=True)))
ans=0
for i in range(0,int(math.sqrt(lim))+1):
    t=str(i*i)
    tmp=[0]*10
    for x in t: tmp[int(x)]+=1
    if len(s)>len(t): tmp[0]+=len(s)-len(t)
    if cnt==tmp: ans+=1
print(ans)

File: AtCoder/ABC/abc324/d/main.nim
import strutils
import algorithm
import math
proc si(): string = stdin.readLine
proc ni(): int = si().parseInt

let _=ni()
var s=si()
var cnt: array[10, int]
for x in s: cnt[ord(x)-ord('0')]+=1
let n=sorted(s,Descending).join("").parseInt
var ans=0
for i in 0..int(sqrt(float64(n))):
    let t=(i*i).intToStr
    var tmp: array[10, int]
    for x in t: tmp[ord(x)-ord('0')]+=1
    if s.len>t.len: tmp[0]+=s.len-t.len
    if cnt==tmp: ans+=1
echo ans